### PR TITLE
TKO-40: BE: Create access code logic for email authentication

### PR DESCRIPTION
**Pull request by MikkalMc**
Ready for review
# Description
Refactored the registration process of the user. This process now consists of two steps:
1) /registerStudentEmail POST via admin
    This post instantiates a new user row in the DB with only email. 
    Generates access code.. in the future the expectation is that we will automatically email the user's
    email provided the generated access code.
    This access code is then used in the next step.
2) /completeRegistration POST via users
    This post is user facing, they will provide the access code, name, email, password to complete the 
    registration process

# Testing
First: you should be able to hit the  /registerStudentEmail with a form body:
- email: test@gmail.com

and see a user be successfully created in the db, retrieve a success response etc etc .
Second: you should be able to hit the /completeRegistration with the form body:
- name: test name
- email: test@gmail.com
- password: ligma
- password_confirmation: ligma
- accessCode: aight this part is different

FOR ACCESS CODE: look into the DB, go to the 'registration_access_code' table and grab the access code there.
# Installation
Run a php artisan migrate:fresh within your docker container

### CODE REVIEW

**Review by alecnicolas — APPROVED** [review]
access code request should only require an email

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 1]
```diff
@@ -13,17 +13,16 @@ class CreateUsersTable extends Migration
...
-                $table->rememberToken();
-                $table->timestamps();
-            });
-        }
+        Schema::create('users', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('name')->default('Student');
+            $table->string('email')->unique();
+            $table->timestamp('email_verified_at')->nullable();
+            $table->string('password')->nullable();
```
Comment (ChanM0) [inline database/migrations/2014_10_12_000000_create_users_table.php:21]: why can the password be nullable?

**Review by ChanM0 — APPROVED** [review]
👍 💯

**Review by ebm17915 — APPROVED** [review]
tested and works!

**Pull request merged**